### PR TITLE
Remake of InstantDummyCaster with vec3 position argument

### PR DESCRIPTION
**Pull request by Trokkin**
It is useful with any kind of projectiles.
I'm in doubt if it is ever needed, but think it's not for me to decide.

### CODE REVIEW

File: wurst/dummy/InstantDummyCaster.wurst [hunk 1]
```diff
@@ -7,111 +7,118 @@ import UnitObjEditing
...
-		..setUnitClassification("_")
-		..setPropulsionWindowdegrees(1.0)
-		..setTooltipBasic("")
-		..setNormalAbilities("Aloc,Agho")
-		..setGroundTexture("")
-		..setShadowTextureBuilding("")
-		..setCollisionSize(0)
-		..setPathingMap("")
-		..setNeutralBuildingShowsMinimapIcon(false)
+	let u = new UnitDefinition(DUMMY_CASTER_UNIT_ID, 'ewsp')
```
Comment (Frotty) [inline wurst/dummy/InstantDummyCaster.wurst]: you don't need `let u =`

File: wurst/dummy/InstantDummyCaster.wurst [hunk 2]
```diff
@@ -7,111 +7,118 @@ import UnitObjEditing
...
 		return castPoint(owner, abilityId, lvl, OrderId(order), targetPos)
-		
+	static function castPoint(player owner, int abilityId, int lvl, string order, vec2 targetPos, vec3 casterPos) returns boolean
+		return castPoint(owner, abilityId, lvl, OrderId(order), targetPos, casterPos)
+	static function castPoint(player owner, int abilityId, int lvl, string order, vec2 targetPos, vec2 casterPos) returns boolean
+		return castPoint(owner, abilityId, lvl, OrderId(order), targetPos, casterPos.toVec3())
+	static function castPoint(player owner, int abilityId, int lvl, int orderId, vec2 targetPos, vec2 casterPos) returns boolean
+		return castPoint(owner, abilityId, lvl, orderId, targetPos, casterPos.toVec3())
+
 constant DUMMY_CASTER_UNIT_ID = 'xcas'
```
Comment (Frotty) [inline wurst/dummy/InstantDummyCaster.wurst]: should probably use compiletime expression with id gen

Comment (Trokkin) [reply wurst/dummy/InstantDummyCaster.wurst]: I guess I know why it wasn't done in the first place... ok, will do.

**Comment by Frotty** [comment]
nice, thanks 🍉

**Pull request merged**